test/sql/aggregate/qualify/test_qualify_macro.test

```diff
@@ -0,0 +1,54 @@
+# name: test/sql/aggregate/qualify/test_qualify_macro.test
+# description: Test QUALIFY clause in the subquery of a macro function
+# group: [qualify]
+
+# load the DB from disk
+load __TEST_DIR__/macro_storage.db
+
+statement ok
+PRAGMA disable_checkpoint_on_shutdown
+
+statement ok
+PRAGMA wal_autocheckpoint='1TB';
+
+statement ok
+CREATE TABLE test (a INTEGER, b INTEGER);
+
+statement ok
+INSERT INTO test VALUES (11, 22), (13, 22), (12, 21)
+
+# create a macro
+statement ok
+CREATE MACRO plus1(x) AS (x + (SELECT COUNT(*) FROM (SELECT b, SUM(test.a) FROM test GROUP BY b QUALIFY row_number() OVER (PARTITION BY b) = 1)))
+
+# use the macro
+query T
+SELECT plus1(3)
+----
+5
+
+statement ok
+DROP MACRO plus1
```

Comment:
The macro storage is not tested since the macro is dropped before the system is restarted. Please alter the test so the system restarts, the macro is used and then the macro is dropped.

Reply:
Great, I will fix it as you said.